src/main/java/uk/gov/companieshouse/service/dissolution/DissolutionService.java

```diff
@@ -63,4 +72,19 @@ public boolean doesDissolutionRequestExistForCompanyByApplicationReference(Strin
     public boolean isDirectorPendingApproval(String companyNumber, String officerId) {
         return getter.isDirectorPendingApproval(companyNumber, officerId);
     }
+
+    public boolean doesDirectorExist(String companyNumber, String officerId) {
+        return getter.doesDirectorExist(companyNumber, officerId);
+    }
+
+    public Optional<String> checkPatchDirectorConstraints(String companyNumber, String directorId, String email) throws DissolutionNotFoundException {
+        final Dissolution dissolution = repository.findByCompanyNumber(companyNumber).orElseThrow(DissolutionNotFoundException::new);
+        if (!getter.isDirectorPendingApprovalForDissolution(directorId, dissolution)) {
+            return Optional.of(DIRECTOR_IS_NOT_PENDING_APPROVAL);
+        }
+        if (!getter.doesEmailBelongToApplicant(email, dissolution)) {
+            return Optional.of(ONLY_THE_APPLICANT_CAN_UPDATE_SIGNATORY);
+        }
+        return Optional.empty();
+    }
```

Comment:
You have increased the size of this class by almost 50% in this one task. Please consider smaller, separate services.

Reply:
Done